### PR TITLE
Saved empty view

### PR DESCRIPTION
**Pull request by eoji**
# what
When users don't have any saved projects, show an empty view with a heart animation. They can also click the heart to see the animation again.

# how
Added `fadeOutAndScale`, `fadeInAndScale`, and `crossFade` to `AnimationUtils`.
Removed some unused drawables.
Updated `DiscoveryFragmentViewModel` to new style with `Inputs` and `Outputs` in the same file.
Updated `DiscoveryFragmentViewModelTest` to have `setUpEnvironment` helper method.

# see

### CODE REVIEW

File: app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt [hunk 1]
```diff
@@ -20,4 +26,52 @@ object AnimationUtils {
     animation.fillAfter = true
     return animation
   }
+
+  @JvmOverloads
+  fun fadeInAndScale(view: View, duration: Long = 500L, startDelay: Long = 500L,
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt]: Open to renaming suggestions.

File: app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt [hunk 2]
```diff
@@ -20,4 +26,52 @@ object AnimationUtils {
...
+                     interpolator: Interpolator = AccelerateDecelerateInterpolator()): ObjectAnimator {
+    val scaleX = PropertyValuesHolder.ofFloat("scaleX", 0f, 1f)
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt]: Should I make these strings constants?

File: app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt [hunk 3]
```diff
@@ -20,4 +26,52 @@ object AnimationUtils {
...
+    return animator
+  }
+
+  @JvmOverloads
+  fun crossFade(visibleView: View, hiddenView: View, crossDuration: Long = 500L, startDelay: Long = 500L,
+                      interpolator: Interpolator = AccelerateDecelerateInterpolator()): AnimatorSet {
+    val crossFadeAnimatorSet = AnimatorSet()
+    val fadeOutAndScale = fadeOutAndScale(visibleView, crossDuration, startDelay, interpolator)
+    val fadeInAndScale = fadeInAndScale(hiddenView, crossDuration, startDelay, interpolator)
+
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt]: What's our new line policy?

File: app/src/main/java/com/kickstarter/viewmodels/DiscoveryFragmentViewModel.java [hunk 4]
```diff
@@ -44,214 +44,295 @@
...
+          this.projectList.onNext(Collections.emptyList());
+        });
+
+      this.paramsFromActivity
+        .compose(combineLatestPair(userIsLoggedIn))
+        .map(pu -> isOnboardingVisible(pu.first, pu.second))
+        .compose(bindToLifecycle())
+        .subscribe(this.shouldShowOnboardingView);
+
+      this.paramsFromActivity
```
Comment (eoji) [inline app/src/main/java/com/kickstarter/viewmodels/DiscoveryFragmentViewModel.java]: Lines 169-181 is what I actually changed.

**Comment by eoji** [comment]
[Update null state for no saved projects (v2)](https://trello.com/c/NJ7p8Hh0/560-update-null-state-for-no-saved-projects-v2)

File: app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt (hunk 1 above)
Comment (Rcureton) [reply app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt]: Do we see this being used elsewhere? If so then the name is fine but if it's just going to be used for the heartbeat animation I would name it something related to it.

File: app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt (hunk 2 above)
Comment (Rcureton) [reply app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt]: Yes seeing that we're using them in many places. Good call 😉

File: app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt (hunk 3 above)
Comment (Rcureton) [reply app/src/main/java/com/kickstarter/libs/utils/AnimationUtils.kt]: For me, as long as it doesn't run the length of the screen it's good with me. What are your thoughts?

**Comment by Rcureton** [comment]
Gif

**Pull request merged**